### ISSUE TITLE
NPC chunk loading issue

### ISSUE THREAD

**Issue by jhooc77** [CitizensDev/Citizens2#2738]
### `/version` output

```txt
[01:37:55 INFO]: This server is running Purpur version git-Purpur-1428 (MC: 1.17.1) (Implementing API version 1.17.1-R0.1-SNAPSHOT) (Git: 9beb23b on ver/1.17.1)
[01:37:55 INFO]: You are running the latest version
```

### `/version Citizens` output

```txt
[01:37:40 INFO]: Citizens version 2.0.29-SNAPSHOT (build 2420)
[01:37:40 INFO]: Website: http://www.citizensnpcs.co
[01:37:40 INFO]: Author: fullwall
```

### Log file

https://paste.denizenscript.com/View/92824

### config.yml

https://paste.denizenscript.com/View/92823

### saves.yml

https://paste.denizenscript.com/View/92822

### Profiler report

https://spark.lucko.me/5peYW6poCZ

### Description

Mspt continued to increase over time.

https://spark.lucko.me/5G0gYWeTdz
About 20 minutes after,
https://spark.lucko.me/5peYW6poCZ

**Comment by mcmonkey4eva** [comment]
- Use Paper, not purpur
- Update your Citizens build that's very behind https://ci.citizensnpcs.co/job/Citizens2/

That lag appears to source from rapid loading/unloading of chunks that contain NPCs.

As a basic workaround, you can enable `/npc chunkload` on all relevant NPCs. That won't necessarily cure the lag depending on what actually causes it, but it will at least remove Citizens NPCs and chunks they're in from the list of variables.

As for the source of the issue - it's hard to say initial, but there are debug options in `Citizens/config.yml`, notably `debug-mode:` and `debug-chunk-loads:` can both be set to true to dump a ridiculous amount of text to your server logs, which can potentially help identify the cause of chunkloading...
though, fair warning, the last couple of times this came up, the chunkloading source has been seemingly just players moving rapidly in and out of range of chunks with NPCs in them (ie player sourced lag, where the fact of it involving Citizens in particular is largely coincidence). In the past, though, we have had cases where the chunkloads were directly caused by Citizens or other plugins and the issue was thus fixable by removing or persisting those chunk loads.

**Comment by jhooc77** [comment]
Yes It is right.
When there were not many people, mspt go up to 40, but when there were 80 people, it didn't even exceed 10.
It's a really strange case.
However, the fundamental problem has still not been solved.
If I understood correctly, should I send you the log that came out after I turned on debug mode?

**Comment by mcmonkey4eva** [comment]
yes

**Comment by jhooc77** [comment]
This profiling data was created when mspt suddenly rose from 10 to 100 while the server was running, maintained for about a minute, and then returned to 10.
https://spark.lucko.me/YwqMBdve7N

One more thing I found today is that when all users were kicked, mspt went up from 10 to 160.
It had been maintained at this value.
Perhaps if left as it is, it could have gone up a little further.



Then I turned on debug mode and opened the server, and even though I didn't even join the server, a huge amount of logs poured out.
https://paste.denizenscript.com/View/92913
